src/lib/alarmMgr/AlarmManager.cpp

```diff
@@ -342,9 +365,9 @@ bool AlarmManager::badInput(const std::string& ip, const std::string& details)
   {
     iter->second += 1;
 
-    if ((badInputLogSampling != 0) && ((badInputs % badInputLogSampling) == 1))
+    if (badInputLogAlways)
     {
-      LM_W(("Repeated BadInput %s: %s", ip.c_str(), details.c_str()));
+      LM_W(("Repeated BadInput for %s: %s", ip.c_str(), details.c_str()));
     }
 
```

Comment:
Reviewing AlarmManager.cpp, the ++ on notificationErrors and badInputs: counter incresing should be done in the "true branch" in badInput() and notificationError().